arangod/Cluster/AgencyCache.cpp

```diff
@@ -512,7 +512,7 @@ bool AgencyCache::registerCallback(std::string const& key, uint64_t const& id) {
   LOG_TOPIC("31415", TRACE, Logger::CLUSTER)
     << "Registered callback for key " << ckey << " with id " << id << ", callbacks: " << size;
   // this method always returns ok.
-  return true;
+  return {};
 }
```

Comment:
Does it make sense to keep a return value at all, if this never returns an error?

Reply:
We could adjust the return value of this method to `void`, but I would like to return a `Result` here, so we can keep return type compatibility between AgencyCache::registerCallback(...) and AgencyCallbackRegistry::registerCallback(...).